### PR TITLE
choreo 1.1.3 (new cask)

### PR DESCRIPTION
**Pull request by CharukaK**
This commit will introduce a new cask which installs choreo cli tool

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

**Comment by bevanjkay** [comment]
This looks like it is failing the notarization audit, however I'm not sure if this should be required.
@krehel Can you speak into this?

**Comment by CharukaK** [comment]
@bevanjkay I will convert this for draft for the moment as I need to bump this to match the latest version

**Comment by krehel** [comment]
Thanks @bevanjkay.

Latest version appears to be validly signed.  The originally submitted version was not signed so the flagging was appropriate (and was also before I made signature check changes)

File: Casks/c/choreo.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    url :url
+    strategy :github_latest
+  end
+
+  binary "choreo"
+
+  zap trash: "~/.choreo"
+
+  # post-install message
+  postflight do
```
Comment (SMillerDev) [inline Casks/c/choreo.rb:27]: This looks like way too much/hard to maintain custom logic to me.

File: Casks/c/choreo.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      completion_script = `#{staged_path}/choreo completion bash`
+    elsif ENV["SHELL"].include? "zsh"
+      if File.exist?("#{user_home}/.zshrc")
+        detected_profile = "#{user_home}/.zshrc"
+      elsif File.exist?("#{user_home}/.zprofile")
+        detected_profile = "#{user_home}/.zprofile"
+      end
+      # Genereate the completion script for the Choreo CLI
+      # save output to a variable
+      completion_script = `#{staged_path}/choreo completion zsh`
```
Comment (SMillerDev) [inline Casks/c/choreo.rb:54]:
```suggestion
      completion_script = Utils.safe_popen_read(staged_path/choreo completion", "zsh")
```

File: Casks/c/choreo.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      elsif File.exist?("#{user_home}/.bashrc")
+        detected_profile = "#{user_home}/.bashrc"
+      elsif File.exist?("#{user_home}/.bash_profile")
+        detected_profile = "#{user_home}/.bash_profile"
+      elsif File.exist?("#{user_home}/.zshrc")
+        detected_profile = "#{user_home}/.zshrc"
+      elsif File.exist?("#{user_home}/.zprofile")
+        detected_profile = "#{user_home}/.zprofile"
+      end
+    end
```
Comment (SMillerDev) [inline Casks/c/choreo.rb:77]: This shouldn't exist in a cask definition

File: Casks/c/choreo.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    if File.exist? detected_profile
+      # Check if the profile file already contains the Choreo CLI completion script
+      if File.foreach(detected_profile).grep(/choreo-completion/).none?
+        File.open(detected_profile, "a") do |file|
```
Comment (SMillerDev) [inline Casks/c/choreo.rb:82]: I reallly don't think we should be reading people's shell config

File: Casks/c/choreo.rb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          file.write("\n# Choreo CLI\n")
+          file.write("[ -f $CHOREO_DIR/bin/choreo-completion ] && source $CHOREO_DIR/bin/choreo-completion")
+        end
+      end
+
+      # Source the profile file
+      system("source", detected_profile.to_s)
```
Comment (SMillerDev) [inline Casks/c/choreo.rb:89]: This won't work since it's not in the user's shell

File: Casks/c/choreo.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
+# Documentation: https://docs.brew.sh/Cask-Cookbook
+#                https://docs.brew.sh/Adding-Software-to-Homebrew#cask-stanzas
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
```
Comment (SMillerDev) [inline Casks/c/choreo.rb:3]: ...

File: Casks/c/choreo.rb (hunk 1 above)
Comment (bevanjkay) [reply Casks/c/choreo.rb:27]: I think the completion scripts should either be pre-generated, or it should just be on the user to generate them after installing the cask.

**Comment by krehel** [comment]
This looks like a 0-star, 0-fork repository that doesn't seem like it meets our inclusion standards.

Also, this cask needs significant rework before it can be considered for acceptance, sorry.

I don't think this should be accepted at the current time.  I'll let other Cask maintainers weigh in.

**Pull request closed without merging**